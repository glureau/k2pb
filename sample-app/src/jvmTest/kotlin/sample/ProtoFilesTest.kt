package sample

import org.approvaltests.Approvals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import java.io.File

@RunWith(JUnit4::class)
class ProtoFilesTest {

    @Test
    fun `check all generated files`() {
        val resourcePath = "build/generated/ksp/jvm/jvmMain/resources/"
        val generatedDir = File(resourcePath)
        println(generatedDir.absolutePath)
        var allMarkdownGenerated = ""
        val files = mutableListOf<File>()
        generatedDir.onEachFile { files += it }
        files.sorted().forEach { file ->
            println("Testing ${file.absolutePath}")
            allMarkdownGenerated += "File: " + file.absolutePath.substringAfter("$resourcePath/") + "\n" + file.readText() + "\n"
        }

        Approvals.verify(allMarkdownGenerated)

        // Should be... but a space is generated 50% of the time from the library ??
        /*
        val files = mutableListOf<File>()
        generatedDir.onEachFile {
            files += it
        }
        CombinationApprovals.verifyAllCombinations({ it.readText().trim() }, files.toTypedArray())
         */
    }

    private fun File.onEachFile(action: (file: File) -> Unit) {
        listFiles().forEach { child ->
            if (child.isDirectory) {
                child.onEachFile(action)
            } else {
                action(child)
            }
        }
    }
}