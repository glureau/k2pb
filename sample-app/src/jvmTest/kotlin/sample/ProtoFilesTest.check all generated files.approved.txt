File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb/ExplicitNullability.proto
syntax = "proto3";

package com.glureau.k2pb;

option java_outer_classname = "K2PBConstants";

// Ensures backward compatibility, when adding a new nullable field, we want to be able to distinguish
// between the absence of the field (old format read) and the default value (ex "" for String).

enum ExplicitNullability {
  // A nullable field has not been explicitly defined, probably an evolution from a previous format.
  // In this case, the associated field will be returned without protobuf default value.
  // Example, a new nullable enum has been added, the default protobuf value would have been
  // technically the first enum entry
  // K2PB will return the associated value without any requirement.
  UNKNOWN = 0;
  
  // A nullable field has been explicitly set to NULL.
  // K2PB ignores the associated field, effectively returning a null for the value.
  NULL = 1;
  
  // A nullable field has been explicitly set to a non-null value.
  // The associated value could be the protobuf default ("" != null).
  // During deserialization, K2PB requires that the associated value is not null, or else throws.
  NOT_NULL = 2;
}
File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/AbstractClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "AbstractClassProto";

import "com/glureau/k2pb_sample/AbstractSubClass.proto";

message AbstractClass {
  oneof abstractClass {
    AbstractSubClass abstractSubClass = 1;
  }
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/AbstractSubClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "AbstractSubClassProto";

message AbstractSubClass {
  int32 foo = 1;
  string bar = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/AnEnum.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_package = "com.glureau.custom.javapackage";
option java_outer_classname = "AnEnumProto";

enum AnEnum {
  AnEnum_A = 0;
  AnEnum_B = 1;
  AnEnum_C = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/BarEvent.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "BarEventProto";

import "com/glureau/k2pb_sample/CommonClass.proto";

message BarEvent {
  CommonClass common = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/BigDecimalHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "BigDecimalHolderProto";

message BigDecimalHolder {
  string bd = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/BigDecimalValueClassHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "BigDecimalValueClassHolderProto";

message BigDecimalValueClassHolder {
  string bdValue = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/CollectionType.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "CollectionTypeProto";

import "com/glureau/k2pb/ExplicitNullability.proto";
import "com/glureau/k2pb_sample/DataClassFromLib.proto";

message CollectionType {
  repeated int32 integerList = 1;
  repeated string stringList = 2;
  repeated int32 maybeIntegerList = 3;
  com.glureau.k2pb.ExplicitNullability isMaybeIntegerListNull = 4;
  map<string, int32> mapStringInt = 5;
  repeated DataClassFromLib dataClassList = 6;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/CommentedClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "CommentedClassProto";

//  This class has kdoc comment.
message CommentedClass {
  //  This field has kdoc comment.
  string fieldWithComment = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/CommonClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "CommonClassProto";

message CommonClass {
  string id = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/CustomNameKtxClasses.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "CustomNameKtxClassesProto";

message CustomNameKtxClasses {
  string instant = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/DataClassFromLib.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_package = "com.glureau.custom.javapackage";
option java_outer_classname = "DataClassFromLibProto";

message DataClassFromLib {
  int32 myInt = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/EnumClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "EnumClassProto";

enum EnumClass {
  A = 0;
  B = 1;
  C = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/EnumHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_package = "com.glureau.custom.javapackage";
option java_outer_classname = "EnumHolderProto";

import "com/glureau/k2pb_sample/AnEnum.proto";

message EnumHolder {
  AnEnum value = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/FooEvent.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "FooEventProto";

import "com/glureau/k2pb_sample/CommonClass.proto";

message FooEvent {
  CommonClass common = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/InlinedCollection.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "InlinedCollectionProto";

import "com/glureau/k2pb_sample/AnEnum.proto";

message InlinedCollection {
  repeated string valueClassList = 1;
  repeated AnEnum valueClassOfEnumList = 2;
  repeated AnEnum valueClassOfNullableEnumList = 3;
  repeated string valueClassOfNullableStringList = 4;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/MultiModule.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "MultiModuleProto";

import "com/glureau/k2pb_sample/DataClassFromLib.proto";

message MultiModule {
  DataClassFromLib dataClassFromLib = 1;
  string valueClassFromLib = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/MyAnnotatedClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "MyAnnotatedClassProto";

message MyAnnotatedClass {
  int32 a = 1;
  int32 b = 4;
  string c = 3;
  string d = 5;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NativeTypeEvent.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NativeTypeEventProto";

message NativeTypeEvent {
  int32 integer = 1;
  int64 long = 2;
  float float = 3;
  double double = 4;
  string string = 5;
  int32 short = 6;
  int32 char = 7;
  bool boolean = 8;
  int32 byte = 9;
  bytes byteArray = 10;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NativeTypeEventUnspecifiedDefault.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NativeTypeEventUnspecifiedDefaultProto";

message NativeTypeEventUnspecifiedDefault {
  int32 integer = 1;
  int64 long = 2;
  float float = 3;
  double double = 4;
  string string = 5;
  int32 short = 6;
  int32 char = 7;
  bool boolean = 8;
  int32 byte = 9;
  bytes byteArray = 10;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableBigDecimalHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableBigDecimalHolderProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableBigDecimalHolder {
  string bd = 1;
  com.glureau.k2pb.ExplicitNullability isBdNull = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableBigDecimalValueClassHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableBigDecimalValueClassHolderProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableBigDecimalValueClassHolder {
  string nullableBdValue = 1;
  com.glureau.k2pb.ExplicitNullability isNullableBdValueNull = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableEnumHolderUnspecifiedDefault.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableEnumHolderUnspecifiedDefaultProto";

import "com/glureau/k2pb/ExplicitNullability.proto";
import "com/glureau/k2pb_sample/AnEnum.proto";

message NullableEnumHolderUnspecifiedDefault {
  AnEnum enum = 1;
  com.glureau.k2pb.ExplicitNullability isEnumNull = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableEnumHolderUnspecifiedNull.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableEnumHolderUnspecifiedNullProto";

import "com/glureau/k2pb/ExplicitNullability.proto";
import "com/glureau/k2pb_sample/AnEnum.proto";

message NullableEnumHolderUnspecifiedNull {
  AnEnum enum = 1;
  com.glureau.k2pb.ExplicitNullability isEnumNull = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableInlinedCollection.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableInlinedCollectionProto";

import "com/glureau/k2pb/ExplicitNullability.proto";
import "com/glureau/k2pb_sample/AnEnum.proto";

message NullableInlinedCollection {
  repeated string valueClassList = 1;
  com.glureau.k2pb.ExplicitNullability isValueClassListNull = 2;
  repeated AnEnum valueClassOfEnumList = 3;
  com.glureau.k2pb.ExplicitNullability isValueClassOfEnumListNull = 4;
  repeated AnEnum valueClassOfNullableEnumList = 5;
  com.glureau.k2pb.ExplicitNullability isValueClassOfNullableEnumListNull = 6;
  repeated string valueClassOfNullableStringList = 7;
  com.glureau.k2pb.ExplicitNullability isValueClassOfNullableStringListNull = 8;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableNativeTypeEvent.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableNativeTypeEventProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableNativeTypeEvent {
  int32 integer = 1;
  com.glureau.k2pb.ExplicitNullability isIntegerNull = 2;
  int64 long = 3;
  com.glureau.k2pb.ExplicitNullability isLongNull = 4;
  float float = 5;
  com.glureau.k2pb.ExplicitNullability isFloatNull = 6;
  double double = 7;
  com.glureau.k2pb.ExplicitNullability isDoubleNull = 8;
  string string = 9;
  com.glureau.k2pb.ExplicitNullability isStringNull = 10;
  int32 short = 11;
  com.glureau.k2pb.ExplicitNullability isShortNull = 12;
  int32 char = 13;
  com.glureau.k2pb.ExplicitNullability isCharNull = 14;
  bool boolean = 15;
  com.glureau.k2pb.ExplicitNullability isBooleanNull = 16;
  int32 byte = 17;
  com.glureau.k2pb.ExplicitNullability isByteNull = 18;
  bytes byteArray = 19;
  com.glureau.k2pb.ExplicitNullability isByteArrayNull = 20;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableNativeTypeEventUnspecifiedDefault.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableNativeTypeEventUnspecifiedDefaultProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableNativeTypeEventUnspecifiedDefault {
  int32 integer = 1;
  com.glureau.k2pb.ExplicitNullability isIntegerNull = 2;
  int64 long = 3;
  com.glureau.k2pb.ExplicitNullability isLongNull = 4;
  float float = 5;
  com.glureau.k2pb.ExplicitNullability isFloatNull = 6;
  double double = 7;
  com.glureau.k2pb.ExplicitNullability isDoubleNull = 8;
  string string = 9;
  com.glureau.k2pb.ExplicitNullability isStringNull = 10;
  int32 short = 11;
  com.glureau.k2pb.ExplicitNullability isShortNull = 12;
  int32 char = 13;
  com.glureau.k2pb.ExplicitNullability isCharNull = 14;
  bool boolean = 15;
  com.glureau.k2pb.ExplicitNullability isBooleanNull = 16;
  int32 byte = 17;
  com.glureau.k2pb.ExplicitNullability isByteNull = 18;
  bytes byteArray = 19;
  com.glureau.k2pb.ExplicitNullability isByteArrayNull = 20;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableNativeTypeEventUnspecifiedNull.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableNativeTypeEventUnspecifiedNullProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableNativeTypeEventUnspecifiedNull {
  int32 integer = 1;
  com.glureau.k2pb.ExplicitNullability isIntegerNull = 2;
  int64 long = 3;
  com.glureau.k2pb.ExplicitNullability isLongNull = 4;
  float float = 5;
  com.glureau.k2pb.ExplicitNullability isFloatNull = 6;
  double double = 7;
  com.glureau.k2pb.ExplicitNullability isDoubleNull = 8;
  string string = 9;
  com.glureau.k2pb.ExplicitNullability isStringNull = 10;
  int32 short = 11;
  com.glureau.k2pb.ExplicitNullability isShortNull = 12;
  int32 char = 13;
  com.glureau.k2pb.ExplicitNullability isCharNull = 14;
  bool boolean = 15;
  com.glureau.k2pb.ExplicitNullability isBooleanNull = 16;
  int32 byte = 17;
  com.glureau.k2pb.ExplicitNullability isByteNull = 18;
  bytes byteArray = 19;
  com.glureau.k2pb.ExplicitNullability isByteArrayNull = 20;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableUuidsHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableUuidsHolderProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableUuidsHolder {
  string uuidAsString = 1;
  com.glureau.k2pb.ExplicitNullability isUuidAsStringNull = 2;
  bytes uuidAsBytes = 3;
  com.glureau.k2pb.ExplicitNullability isUuidAsBytesNull = 4;
  string stringValueClass = 5;
  com.glureau.k2pb.ExplicitNullability isStringValueClassNull = 6;
  bytes bytesValueClass = 7;
  com.glureau.k2pb.ExplicitNullability isBytesValueClassNull = 8;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/NullableValueClassHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "NullableValueClassHolderProto";

import "com/glureau/k2pb/ExplicitNullability.proto";

message NullableValueClassHolder {
  string valueClassFromLib = 1;
  com.glureau.k2pb.ExplicitNullability isValueClassFromLibNull = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/ObjectClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "ObjectClassProto";

message ObjectClass { }

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/StandardClass.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "StandardClassProto";

message StandardClass {
  string eventUUID = 1;
  bytes bytes = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/TransientField.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "TransientFieldProto";

message TransientField {
  string fieldSerialized = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/User.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "UserProto";

import "com/glureau/k2pb_sample/Vehicle.proto";

message User {
  string name = 1;
  Vehicle vehicle = 2;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/UuidsHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "UuidsHolderProto";

message UuidsHolder {
  string uuidAsString = 1;
  bytes uuidAsBytes = 2;
  string stringValueClass = 3;
  bytes bytesValueClass = 4;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/ValueClassList.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_package = "com.glureau.custom.javapackage";
option java_outer_classname = "ValueClassListProto";

message ValueClassList {
  repeated string valueClassFromLibs = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/ValueClassOfEnumHolder.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_package = "com.glureau.custom.javapackage";
option java_outer_classname = "ValueClassOfEnumHolderProto";

import "com/glureau/k2pb_sample/AnEnum.proto";

message ValueClassOfEnumHolder {
  AnEnum value = 1;
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/Vehicle.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "VehicleProto";

// Polymorphism structure for 'Vehicle'
// Possible values are:
// - 'Vehicle.Bike'
// - 'Vehicle.Car'
message Vehicle {
  oneof vehicle {
    Vehicle.Bike bike = 1;
    Vehicle.Car car = 2;
  }
  message Bike {
    string brand = 1;
  }
  message Car {
    string brand = 1;
  }
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/WithNestClassA.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "WithNestClassAProto";

message WithNestClassA {
  WithNestClassA.NestedClass a = 1;
  message NestedClass {
    string nested = 1;
  }
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/WithNestClassB.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "WithNestClassBProto";

message WithNestClassB {
  WithNestClassB.NestedClass b = 1;
  message NestedClass {
    WithNestClassB.NestedClass.NestedEnum nestedEnum = 1;
    enum NestedEnum {
      A = 0;
      B = 1;
      C = 2;
    }
  }
}

File: /Users/gregorylureau/Projects/k2pb/sample-app/build/generated/ksp/jvm/jvmMain/resources/k2pb/com/glureau/k2pb_sample/WithNestedEnum.proto
syntax = "proto3";

package com.glureau.k2pb_sample;

option java_outer_classname = "WithNestedEnumProto";

message WithNestedEnum {
  WithNestedEnum.NestedEnum enum = 1;
  enum NestedEnum {
    A = 0;
    B = 1;
    C = 2;
  }
}

